tests/Fixer/StringNotation/HeredocClosingMarkerFixerTest.php

```diff
@@ -0,0 +1,284 @@
+                <?php $a = B<<<EOD
+                xxx EOD xxx
+                EOD;
+
+                EOF,
+            <<<'EOF'
+                <?php $a = B<<<TEST
+                xxx EOD xxx
+                TEST;
+
+                EOF,
+        ];
+
+        yield 'heredoc and reserved closing marker' => [
+            <<<'EOF'
+                <?php $a = <<<PHP
+                xxx
+                PHP;
+
+                EOF,
+        ];
+
+        yield 'heredoc and reserved closing marker - different case' => [
+            <<<'EOF'
+                <?php $a = <<<PHP
+                xxx
+                PHP;
+                $a = <<<PHP
+                PHP;
+
+                EOF,
+            <<<'EOF'
+                <?php $a = <<<php
+                xxx
+                php;
+                $a = <<<Php
+                Php;
+
+                EOF,
+        ];
+
+        yield 'heredoc and reserved custom closing marker' => [
+            <<<'EOF'
+                <?php $a = <<<Žlutý
+                xxx
+                Žlutý;
+                $aNormCase = <<<Žlutý
+                xxx
+                Žlutý;
+                $b = <<<EOD
+                xxx2
+                EOD;
+                $c = <<<EOD
+                xxx3
+                EOD;
+
+                EOF,
+            <<<'EOF'
+                <?php $a = <<<Žlutý
+                xxx
+                Žlutý;
+                $aNormCase = <<<ŽlUTý
+                xxx
+                ŽlUTý;
+                $b = <<<Žlutý2
+                xxx2
+                Žlutý2;
+                $c = <<<PHP
+                xxx3
+                PHP;
+
+                EOF,
+            ['reserved_closing_markers' => ['Žlutý']],
+        ];
+
+        yield 'heredoc /w content starting with preferred closing marker' => [
+            <<<'EOF'
+                <?php $a = <<<EOD_
+                EOD xxx
+                EOD_;
+
+                EOF,
+            <<<'EOF'
+                <?php $a = <<<TEST
+                EOD xxx
+                TEST;
+
+                EOF,
+        ];
+
+        yield 'heredoc /w content starting with whitespace and preferred closing marker' => [
+            <<<'EOF'
+                <?php $a = <<<EOD_
+                 EOD xxx
+                EOD_;
+
+                EOF,
+            <<<'EOF'
+                <?php $a = <<<TEST
+                 EOD xxx
+                TEST;
+
+                EOF,
+        ];
+
+        yield 'heredoc /w content starting with preferred closing marker and single quote' => [
+            <<<'EOF'
+                <?php $a = <<<EOD_
+                EOD'
+                EOD_;
+
+                EOF,
+            <<<'EOF'
+                <?php $a = <<<TEST
+                EOD'
+                TEST;
+
+                EOF,
+        ];
+
+        yield 'heredoc /w content starting with preferred closing marker and semicolon' => [
+            <<<'EOF'
+                <?php $a = <<<EOD_
```

Comment:
see https://3v4l.org/98K7e - automatic conflict resolution, very powerful, as then all heredocs can be fixed, even the one with otherwise conflicting markers, nevertheless such conflicts are very rare in real strings

Reply:
Thanks for the explanation 🙂.